### ISSUE TITLE
Help does not display if the argument contains an underscore

### ISSUE THREAD

**Issue by bmamouri** [pyinvoke/invoke#533]
Pretty much the title. Here is how to reproduce it:

```
@task(help={
    'service': "The service that should be deployed",
    'env': 'Target environment',
    'git_branch': '[Optional] Name of the branch to be used in the host machines',
})
def deploy(c, service, env, git_branch=None):
    """
    Deploy a service into a specific environment and optionally using a specific git branch
    """
```
The result of `invoke -h deploy`:

```
Options:
  -e STRING, --env=STRING          Target environment
  -g STRING, --git-branch=STRING
  -s STRING, --service=STRING      The service that should be deployed
```

Also, I have no idea why the options are not listed in the right order!

**Comment by bmamouri** [comment]
It seems optional argument that has an underscore in them are not recognized:

```
@task(optional=['git_branch'])
def deploy(c, service, env, git_branch=None):
    print(git_branch)
```

The output:
```
No idea what '--git_branch' is!
```

If having underscore is not allowed in variable names, we need to show a more meaningful error.

**Comment by bitprophet** [comment]
I think this is a disjoint between the Python and CLI side of things, brought on by the automatic underscores-to-dashes conversion. I suspect just changing the strings in `@task` args, to have dashes instead of underscores, may be the fastest workaround? Haven't tried myself yet.

Ideally we'll patch things so that's not necessary and either A) your expectation gets met and we explicitly set things up so references to args use the Python-side identifiers w/ underscores; or B) we make it so either one works. Those changes probably look very similar either way.

----

Re: "the right order" - let's unpack that :) why do you assume the right order is the order listed in the function, and not alphabetical order (which is, IIRC, what is currently happening)? In my experience, when CLI applications do any sorting whatsoever, they sort alphabetically so users looking at `--help` have an easier time finding a given flag.

In a vacuum, I'd suspect that argument order _may_ be more intuitive for the author of the task, but that alpha order is typically more intuitive for consumers/users. Curious what others think here.

**Comment by bitprophet** [comment]
Duplicate of #409 and should be fixed when I merge #611.
